clients/da-vinci-client/src/main/java/com/linkedin/davinci/storage/StorageService.java

```diff
@@ -178,6 +179,21 @@ private void initInternalStorageEngineFactories() {
     persistenceTypeToStorageEngineFactoryMap.put(BLACK_HOLE, new BlackHoleStorageEngineFactory());
   }
 
+  private boolean deleteStorageEngine(String storageEngineName) {
+    String storeName = Version.parseStoreFromKafkaTopicName(storageEngineName);
+    Store store;
+    try {
+      store = storeRepository.getStoreOrThrow(storeName);
+    } catch (VeniceNoStoreException e) {
+      // The store does not exist in Venice anymore, so it will be deleted.
+      LOGGER.warn("Store does not exist, will delete invalid local version: {}", storageEngineName);
+      return true;
+    }
+
+    int versionNumber = Version.parseVersionFromKafkaTopicName(storageEngineName);
+    return !store.getVersion(versionNumber).isPresent() || versionNumber < store.getCurrentVersion();
```

Comment:
So if we remove the backup version, it means Server will re-bootstrap the deleted backup version, right?

Reply:
yes, but that could help in successful re-ingestion, as this case is only for error path.